0.19.0 blogpost (#145)
TeamNewPipe/website: _posts/2020-03-29-newpipe-0.19.0-released.md

---
layout: post
title: "NewPipe 0.19.0 released: Folder your feeds and mute them"
short: "NewPipe 0.19.0 released"
date: 2020-03-29 00:00:00 +0100
categories: [release, pinned]
author: Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

> A new day, a new full release. In today's update, NewPipe is proudly bringing you a great way to sort your subscriptions and a mute button, on top of the usual smaller improvements and fixes<!-- more -->

Reviewer: ```suggestion
A new day, a new full release. In today's update, NewPipe is proudly bringing you a great way to sort your subscriptions, and a mute button, on top of the usual smaller improvements and fixes<!-- more -->
```

Author: the and is a listing here of only two items, the comma would be incorrect

Reviewer: There is ambiguity, though. You could change it to "bringing you a mute button and a great way to sort your subscriptions".

Though I think that should be "manage your subscribed channels" just to be clear.

Author: "bringing you a mute button and a great way to manage your subscribed channel".

sounds good to me

## Folders

> Since moving images say more then my words, please have a look at them:

Reviewer 2: then->than

<video src="/img/folders.mp4" class="no-flow img-responsive" width="220" height="140" autoplay loop></video>

> For everyone who rather enjoys reading, this is for you: NewPipe now supports grouping subscriptions into folders. All the recent videos of those channels then appear in one re-loadable feed when you press on the folder. While [@mauriciocolli](https://github.com/mauriciocolli) implemented that, he also improved the `What's New` sidebar feed. And on top of that, live streams have now a viewer count in those feeds!

Reviewer 2: For everyone who rather enjoys reading->For those who prefer to read

The other wording sounds.... off

Author: I don't prefer any version, so will enjoy using yours.

> For everyone who rather enjoys reading, this is for you: NewPipe now supports grouping subscriptions into folders. All the recent videos of those channels then appear in one re-loadable feed when you press on the folder. While [@mauriciocolli](https://github.com/mauriciocolli) implemented that, he also improved the `What's New` sidebar feed. And on top of that, live streams have now a viewer count in those feeds!

Reviewer 3: I was first suprised you that you haven't talked about **fixed feed** in title, but even more surprised later by the fact you didn't talk about feed **being finally fixed**, showing **in the right order**, and not randomly throwing the first video of each channel in this paragraph.
Also, I think that you should give a small word on _fast mode_, or _fetch from a dedicated feed_ vs _slow mode_, or _not fast mode_, I don't know how to call this. Just explain quickly the difference, [and this string should help you](https://github.com/TeamNewPipe/NewPipe/blob/505a689268d5a08b16624016f120a36e97d5976c/app/src/main/res/values/strings.xml#L641).

> For those who prefer to read, this is for you: NewPipe now supports grouping subscriptions into folders. All the recent videos of those channels then appear in one re-loadable feed when you press on the folder. While [@mauriciocolli](https://github.com/mauriciocolli) implemented that, he also improved the `What's New` sidebar feed. With improved I mean finally fixing it: The videos are showing in the correct order of the upload, not the first video each channel in a random order. And on top of that, live streams have now a viewer count in those feeds!

Reviewer: ```suggestion
For those who prefer to read, this is for you: NewPipe now supports grouping subscriptions into folders. All the recent videos of the channels in that folder then appear in one feed which can be refreshed when you press on the folder. While [@mauriciocolli](https://github.com/mauriciocolli) implemented that, he also improved the `What's New` sidebar feed. By improved, I mean that he finally fixed it: The videos now show in the correct order, by the upload date and time, not the first videos of each channel in a random order like before. And on top of that, live streams now have a viewer count in those feeds!
```

An additional feature squashed into this change is the implementation of a so called _fast-mode_. If you enable it in the `Content` menu, those feeds will load much faster from the services which provide a so called _dedicated feed_. The downside of that is that those feeds contain less information, which might result in duration or the live status of videos not shown in the preview. The decision is up to you: Living on the speeding edge and missing information or taking it nice and slow, but knowing **everything**.

## Mute button

> This release doesn't stop at folders though. Thanks to [@karkaminski](https://github.com/karkaminski), NewPipe now has mute button. [@B0pol](https://github.com/B0pol) formed it further into the shape you can experience it in now. _"But where can I find it"_ you ask. Fear not, for I have prepared a gallery of screenshots. To access the pop-up and background player menus shown, you need to press on the notification from NewPipe.

Reviewer: ```suggestion
This release doesn't stop at folders though. Thanks to [@karkaminski](https://github.com/karkaminski), NewPipe now has a mute button. [@B0pol](https://github.com/B0pol) formed it further into the shape you can experience it in now. _"But where can I find it?"_ you ask. Fear not, for I have prepared a gallery of screenshots. To access the pop-up and background player menus shown, you need to press on the notification from NewPipe.
```

<img class="no-flow vertical" src="/img/screenshots/mute_main.jpg"/>
<img class="no-flow img-responsive" src="/img/screenshots/mute_back.jpg"/>

## Smaller Improvements

> [@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@XiangRongLins](https://github.com/XiangRongLin) goal, he made sure that the seek duration jumped visibly to an even one when the fast inexact search option gets activated. Previously the user wouldn't see this directly, but experience when seeking for- or backwards. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings moved from `Appearance` to `Content`, where they belonged all along: `Show 'Next'` and `Content of main page`.

Reviewer: ```suggestion
[@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@XiangRongLin's](https://github.com/XiangRongLin) goal. He made sure that the seek duration jumped visibly to an even one when the fast inexact search option gets activated. Previously the user wouldn't see this directly, but experience it when seeking forwards or backwards. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings have been moved from `Appearance` to `Content`, where they belonged all along: `Show 'Next'` and `Content of main page`.
```

> [@wb9688](https://github.com/wb9688) and [@B0pol](https://github.com/B0pol) (in his second contribution) added support for more subdomains. Now `music.youtube.com` and `media.ccc.de` are natively supported by NewPipe, thanks you two. In the last point to this section, [@jonas-skywalker](https://github.com/jonas-skywalker) (may the force be with you) improved the search section. Now the magnifying glass vanishes once you start typing.

Reviewer: ```suggestion
[@wb9688](https://github.com/wb9688) and [@B0pol](https://github.com/B0pol) (in his second contribution) added support for more subdomains. Now `music.youtube.com` and `media.ccc.de` are natively supported by NewPipe. Thanks, you two. In the last point in this section, [@jonas-skywalker](https://github.com/jonas-skywalker) (may the force be with you) improved the search section. Now the magnifying glass vanishes once you start typing.
```

Reviewer 4: media.ccc.de was supported before (and it is no subdomain, it's a whole service). New in this version is that one can share their URLs to NewPipe.

Author: oh okay.

Reviewer 4: technically yes, but ccc.de is not a streaming / media service :D

Reviewer: But MediaCCC is a distinct Service Newpipe supports. So whether or not the URL is a subdomain is not only irrelevant here, but also a bit misleading. I think there is not need to use the term at all. It's enough to say that MediaCCC and Youtube Music URLs can now be shared to Newpipe without error.

Author: I dropped subdomain now, is that alright?

Reviewer 4: Yes. Thanks

[@wb9688](https://github.com/wb9688) and [@B0pol](https://github.com/B0pol) (in his second contribution) added support for more subdomains. Now `music.youtube.com` and `media.ccc.de` are natively supported by NewPipe, thanks you two. In the last point to this section, [@jonas-skywalker](https://github.com/jonas-skywalker) (may the force be with you) improved the search section. Now the magnifying glass vanishes once you start typing.

## Fixes a.k.a. NewPipe crashes less \o/

> Andriod 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The app also crashed when you tried to download a really large video, which [@XiangRongLin](https://github.com/XiangRongLin) fixed in his third contribution. The seek duration setting was not properly shown in any other language then English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bugs, which made some WebM videos non seekable. Now you can seek all you want.

Reviewer 2: Andriod->Android

> Android 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The app also crashed when you tried to download a really large video, which [@XiangRongLin](https://github.com/XiangRongLin) fixed in his third contribution. The seek duration setting was not properly shown in any other language then English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bugs, which made some WebM videos non seekable. Now you can seek all you want.

Reviewer 5: > The app also crashed when you tried to download a really large video, which [@XiangRongLin](https://github.com/XiangRongLin) fixed in his third contribution

I don't remember doing this. I think you mixed it up with someone else

Reviewer 4: Correct. This was kapodamy

Author: grrr. Reality should be what I want.

> Android 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The app also crashed when you tried to download a really large video, which [@XiangRongLin](https://github.com/XiangRongLin) fixed in his third contribution. The seek duration setting was not properly shown in any other language then English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bugs, which made some WebM videos non seekable. Now you can seek all you want.

Reviewer: ```suggestion
Android 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The app also crashed when you tried to download a really large video, which [@XiangRongLin](https://github.com/XiangRongLin) fixed in his third contribution. The seek duration setting was not properly shown in any other language then English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bug, which made some WebM videos non seekable. Now you can seek all you want.
```

> All of the following fixes were done by the incredible [@mauriciocolli](https://github.com/mauriciocolli). Overall, he brought seven (!) different contributions in todays release. He solved an issue where the last screen of the app could be popping up during the app exit animation. A bug in the main screen resulted to tabs not being restored correctly if the app was put to sleep by Android. The main screen had another issue, especially with the menu, which could vanish until the user interacted with the app again. Sharing a YouTube link to NewPipe could end up crashing NewPipe if certain characters were in those links, not an issue anymore. And finally, he improved the handling of unavailable content on YouTube, which resulted in a crash previously.

Reviewer: ```suggestion
All of the following fixes were done by the incredible [@mauriciocolli](https://github.com/mauriciocolli). Overall, he brought seven (!) different contributions to today's release. He solved an issue where the last screen of the app could be popping up during the app exit animation. A bug in the main screen resulted in tabs not being restored correctly if the app was put to sleep by Android. The main screen had another issue, especially with the menu, which could vanish until the user interacted with the app again. Sharing a YouTube link with NewPipe could end up crashing NewPipe if certain characters were in those links, which is not an issue anymore. And finally, he improved the handling of unavailable content on YouTube, which resulted in a crash previously.
```

## Development

This release also includes two big improvements of the development process. [@Stypox](https://github.com/Stypox) and [@mauriciocolli](https://github.com/mauriciocolli) changed the way `.apk` files are generated in the development process. This makes it way easier to test a change on your device, since you can have all of these NewPipe apps alongside each other, as long as you have the storage for them.

[@Poolitzer](https://github.com/poolitzer), who already took over the latest blog posts, now created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. The final form of them that you are experiencing is the result of a combined effort of the whole community, thanks for everyone who forced me to implement and discuss their suggestions.

## Conclusion

> That’s it for this release. A big thanks goes out to all the people who made this possible, especially those not mentioned: Developers for helping out fellow developers, testers for testing, maintainers for maintaining, users like you for reporting the bugs in the first place. If you'd like to get a more technical view into the app, you can find the release page [here](https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.19.0) and the corresponding PR [here](https://github.com/TeamNewPipe/NewPipe/pull/3267).

Reviewer 3: Press F to pay respect for translators

Author: did we had big changes in this release for them? I only saw you messing around with plural strings when you fixed the second duration bug

Reviewer 3: Look at commits from weblate: https://github.com/TeamNewPipe/NewPipe/pull/3267
And yeah, something big happened, **Nepali has been fully translated**, look at this huge commit https://github.com/TeamNewPipe/NewPipe/pull/3267/commits/373ec960513a9eac5a1b47f5141635e7fe836221

Anyway, I don't see any reason not to mention them when the sentence is about thanking the other contributors not mentionned in the post.

Author: Hm, I mention them then

> [@Poolitzer](https://github.com/poolitzer), who already took over the latest blog posts, now created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. The final form of them that you are experiencing is the result of a combined effort of the whole community, thanks for everyone who forced me to implement and discuss their suggestions.

Reviewer: ```suggestion
[@Poolitzer](https://github.com/poolitzer) (that's me!), who already took over the latest blog posts, created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. Their final form that you are experiencing is the result of the combined effort of the whole community, thanks to everyone who forced him to implement and discuss their suggestions.
```

Author: I love the thats me part, but why change the last me to him? we could just stay with me imo

Reviewer: Because you're talking about yourself in third person at the beginning. If you want to make the whole thing first person, then do something like:

"I (Poolitzer) had earlier taken over making the blog posts. Now I have created up to date..."

Author: hmm true. Maybe I should just delete that half sentence, doesnt make much sense their.

Reviewer: Alright. I'll review it after your changes.

Author: Tada

---
layout: post
title: "NewPipe 0.19.0 released: Folder your feeds and mute them"
short: "NewPipe 0.19.0 released"
date: 2020-03-29 00:00:00 +0100
categories: [release, pinned]
author: Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

A new day, a new full release. In today's update, NewPipe is proudly bringing you a great way to sort your subscriptions and a mute button, on top of the usual smaller improvements and fixes<!-- more -->

> ## Folders and fixed Feeds

Reviewer 4: We should be consistent here and call this feature "Feed Groups" as it is called in the app.

Reviewer 4: Any thoughts on this? Do you agree / disagree?

Reviewer: "WHAT'S NEW FEED IS FIXED" should definitely be highlighted very much. "Feed order and subscription groups" is the name of the PR, and I think that is a suitable title.

Actually, even though they are part of one PR, they should be separate headings. Maybe:

## What's New now shows the latest videos first! (Finally!)

and

## Subscription Groups

Author: > What's New now shows the latest videos first! (Finally!)

if I put this in its own heading, it wouldnt get much content. Rn its like two sentences

Reviewer: It's a major pet peeve of users (going by issue comments on Github, as well as personal feedback from friends). It should be highlighted strongly in some way, if not its own heading.

Author: okay, it has its own heading "now"

## Mute button

This release doesn't stop at folders though. Thanks to [@karkaminski](https://github.com/karkaminski), NewPipe now has mute button. [@B0pol](https://github.com/B0pol) formed it further into the shape you can experience it in now. _"But where can I find it"_ you ask. Fear not, for I have prepared a gallery of screenshots. To access the pop-up and background player menus shown, you need to press on the notification from NewPipe.

<img class="no-flow vertical" src="/img/screenshots/mute_main.jpg"/>
<div class="media-row media-items-2">
<img class="no-flow img-responsive" src="/img/screenshots/mute_pop.jpg"/>
<img class="no-flow img-responsive" src="/img/screenshots/mute_back.jpg"/>
</div>

## Smaller Improvements

> [@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@XiangRongLins](https://github.com/XiangRongLin) goal, he made sure that the seek duration jumped visibly to an even one when the fast inexact search option gets activated. Previously the user wouldn't see this directly, but experience when seeking for- or backwards. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings moved from `Appearance` to `Content`, where they belonged all along: `Show 'Next'` and `Content of main page`.

Reviewer 4: [@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@**XiangRongLin's**](https://github.com/XiangRongLin) goal, **he** _(he or she? I am not sure whether Xiang or Lin is the forename here [AFAIK Xiang can be used by males and females and I've had a classmate and she was called Lin])_ made sure that the seek duration jumped visibly to an even one when the fast inexact search option gets activated. Previously the user wouldn't see this directly, but experience when seeking for- or backwards. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings moved from `Appearance` to `Content`, where they belonged all along: `Show 'Next'` and `Content of main page`.

Reviewer 3: If you don't know, you can word the sentence differently. Here, who fits good imo: ``Further improving the user experience was also @XiangRongLin's goal, **who** made sure that the seek duration jumped visibly to an even one when the fast inexact search option gets activated.

Reviewer 4: I think @B0pol's suggestions is good

Reviewer 5: Previously the user could get "stuck" when seeking in the main video with 5 second seek duration and inexact seek active. That was the main reason for making the seek duration jump to the next "valid" value when activating inexact seek.

See TeamNewPipe/NewPipe#3127

---
layout: post

> title: "NewPipe 0.19.0 released: Group your feeds and mute them"

Reviewer: We're grouping channels _into_ feeds, aren't we?
```suggestion
title: "NewPipe 0.19.0 released: Group your channels and mute your videos"
```

short: "NewPipe 0.19.0 released"
date: 2020-03-29 00:00:00 +0100
categories: [release, pinned]
author: Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

> A new day, a new full release. In today's update, NewPipe is proudly bringing you a mute button and a great way to manage your subscribed channel, on top of the usual smaller improvements and fixes.<!-- more -->

Reviewer: ```suggestion
A new day, a new full release. In today's update, NewPipe is proudly bringing you a mute button and a great way to manage your subscribed channels, on top of the usual smaller improvements and fixes.<!-- more -->
```

> ## Feed groups

Reviewer: I know it says Feed Groups in the app itself, but I think calling them Channel Groups will be clearer, and the string should be changed in the app too. We're grouping channels _into_ feeds. @TobiGr
```suggestion
## Channel Groups
```

Reviewer 4: >  We're grouping channels **into** feeds.

That's a good point. However, we can only change the app strings in the next update. They are already translated into several languages. Can you please open a PR / an issue to discuss this in the app repo or on IRC?

Reviewer: No problem. I'll open an issue, not a PR, since B0pol is the expert on translations.

Author: So Im not going to merge the suggestion

Reviewer: Yeah, let's leave this for the next one.

Since moving images say more than my words, please have a look at them:

<video src="/img/folders.webm" class="no-flow img-responsive" width="220" height="140" autoplay loop></video>

> For those who prefer to read, this is for you: NewPipe now supports grouping subscriptions into folders. All the recent videos of those channels then appear in one re-loadable feed when you press on the folder. We have to thank [@mauriciocolli](https://github.com/mauriciocolli) for that.

Reviewer: ```suggestion
For those who prefer to read, this is for you: NewPipe now supports grouping subscribed channels into feeds. All the recent videos of those channels can be seen at once when you refresh the feed. We have [@mauriciocolli](https://github.com/mauriciocolli) to thank for that.
```

> An additional feature squashed into this change is the implementation of a so called _fast-mode_. If you enable it in the <code><nobr><img src="/img/icons/baseline-language-20px.svg" /> Content</nobr></code> settings, those feeds will load much faster from the services which provide a so called _dedicated feed_. The downside of that is that those feeds contain less information, which might result in duration or the live status of videos not shown in the preview. The decision is up to you: Living on the speeding edge and missing information or taking it nice and slow, but knowing **everything**.

Reviewer: ```suggestion
An additional feature squashed into this change is the implementation of a so called _fast-mode_. If you enable it in the <code><nobr><img src="/img/icons/baseline-language-20px.svg" /> Content</nobr></code> settings, those feeds will load much faster from the services which provide a _dedicated feed_. The downside is that those feeds contain less information, which might result in the duration or the live status of videos not being shown in the preview. The decision is up to you: Living on the speeding edge and missing information or taking it nice and slow, but knowing **everything**.
```

## What's New now shows the latest videos first! (Finally!)

> Part of the above feature is also the now improved `What's New` sidebar feed. With improved I mean finally a fixed version: The videos are showing in the correct order of the upload, not the first video each channel in a random order. And on top of that, live streams have now a viewer count in those feeds!

Reviewer: ```suggestion
Part of the above feature is also the much improved `What's New` sidebar feed. By improved I mean fixed, finally: The videos now show in the correct order of uploading, whereas the previous version showed the first video of each channel in a random order (roughly in the order you subscribed to them). And on top of that, live streams now have a viewer count in those feeds!
```

Author: the header already says finally, kinda doubled...

Reviewer: I just moved it from before to after "fixed". Removing it is fine.

Author: but now it stands out that much :D

## Mute button

> This release doesn't stop at folders though. Thanks to [@karkaminski](https://github.com/karkaminski), NewPipe now has a mute button. [@B0pol](https://github.com/B0pol) formed it further into the shape you can experience it in now. _"But where can I find it?"_ you ask. Fear not, for I have prepared a gallery of screenshots. To access the pop-up and background player menus shown, you need to press on the notification from NewPipe.

Reviewer: ```suggestion
This release doesn't stop at channel groups, though. Thanks to [@karkaminski](https://github.com/karkaminski), NewPipe now has a mute button. [@B0pol](https://github.com/B0pol) molded it further into the shape you can experience it in now. _"But where can I find it?"_ you ask. Fear not, for I have prepared a gallery of screenshots. To access the pop-up and background player menus shown, you need to press on the notification from NewPipe.
```

<img class="no-flow vertical" src="/img/screenshots/mute_main.jpg"/>
<img class="no-flow img-responsive" src="/img/screenshots/mute_back.jpg"/>

## Smaller Improvements

> [@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@XiangRongLin's](https://github.com/XiangRongLin) goal, who made sure user do not get "stuck" when seeking in the main video player with 5 second seek duration and inexact seek enabled. Previously user could not get back when hitting the rewind button. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings moved from <code><nobr><img src="/img/icons/color_lens-20px.svg" /> Appearance</nobr></code> to <code><nobr><img src="/img/icons/baseline-language-20px.svg" /> Content</nobr></code>, where they belonged all along: `Show 'Next'` and `Content of main page`.

Reviewer: ```suggestion
[@comradekingu](https://github.com/comradekingu) took care of the English base strings in this release and made sure they are consistent. Further improving the user experience was also [@XiangRongLin's](https://github.com/XiangRongLin) goal, who made sure users do not get "stuck" when seeking in the main video player with seek duration set to 5 seconds and inexact seek enabled. Previously users could not seek backwards when hitting the rewind button. Thanks to [@opusforlife2](https://github.com/opusforlife2), two settings moved from <code><nobr><img src="/img/icons/color_lens-20px.svg" /> Appearance</nobr></code> to <code><nobr><img src="/img/icons/baseline-language-20px.svg" /> Content</nobr></code>, where they belonged all along: `Show 'Next'` and `Content of main page`.
```

[@wb9688](https://github.com/wb9688) and [@B0pol](https://github.com/B0pol) (in his second contribution) added share-to-NewPipe support for more URLs. Now `music.youtube.com` and `media.ccc.de` are natively supported. Thanks, you two. In the last point in this section, [@jonas-skywalker](https://github.com/jonas-skywalker) (may the force be with you) improved the search section. Now the magnifying glass vanishes once you start typing.

## Fixes a.k.a. NewPipe crashes less \o/

> Android 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The seek duration setting was not properly shown in any other language then English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bug, which made some WebM videos non seekable. Now you can seek all you want. He also fixed a crash when you tried to download a really large video.

Reviewer: ```suggestion
Android 9 changed how databases behave, effectively destroying backups and exporting remote playlists - not anymore, thanks to [@yausername](https://github.com/yausername). [@XiangRongLin](https://github.com/XiangRongLin) improved sharing a downloaded video to other apps, which crashed the app before this release. The seek duration setting was not properly shown in any other language than English, but [@B0pol](https://github.com/B0pol) let them magically appear again. And finally, [@kapodamy](https://github.com/kapodamy) took care of a nasty bug, which made some WebM videos non seekable. Now you can seek all you want. He also fixed a crash when you tried to download a really large video.
```

Reviewer: Wait, why isn't Github showing the changes in red and green highlights here?

All of the following fixes were done by the incredible [@mauriciocolli](https://github.com/mauriciocolli). Overall, he brought seven (!) different contributions to today's release. He solved an issue where the last screen of the app could be popping up during the app exit animation. A bug in the main screen resulted in tabs not being restored correctly if the app was put to sleep by Android. The main screen had another issue, especially with the menu, which could vanish until the user interacted with the app again. Sharing a YouTube link with NewPipe could end up crashing NewPipe if certain characters were in those links, which is not an issue anymore. And finally, he improved the handling of unavailable content on YouTube, which resulted in a crash previously.

## Development

> This release also includes two big improvements of the development process. [@Stypox](https://github.com/Stypox) and [@mauriciocolli](https://github.com/mauriciocolli) changed the way `.apk` files are generated in the development process. This makes it way easier to test a change on your device, since you can have all of these NewPipe apps alongside each other, as long as you have the storage for them.

Reviewer: ```suggestion
This release also includes two big improvements in the development process. [@Stypox](https://github.com/Stypox) and [@mauriciocolli](https://github.com/mauriciocolli) changed the way `.apk` files are generated in the development process. This makes it way easier to test a change on your device, since you can install all of these test versions of Newpipe alongside each other, as long as you have the storage for them.
```

> [@Poolitzer](https://github.com/poolitzer) (that's me!) created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. Their final form that you are experiencing is the result of a combined effort of the whole community, thanks for everyone who forced me to implement and discuss their suggestions, especially [@opusforlife2](https://github.com/opusforlife2) and [@B0pol](https://github.com/B0pol).

Reviewer: ```suggestion
[@Poolitzer](https://github.com/poolitzer) (that's me!) created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. Their final form that you are experiencing is the result of the combined effort of the whole community, thanks to everyone whose suggestions were discussed and implemented, especially [@opusforlife2](https://github.com/opusforlife2) and [@B0pol](https://github.com/B0pol).
```

Author: dropping the force me part here, I liked it :(. gave the (that's me!) part a sense to its existence

Reviewer 4: I agree, let's do this
```suggestion
[@Poolitzer](https://github.com/poolitzer) (that's me!) created up-to-date issue templates, which makes it easier for you to report a bug or suggest a feature. Their final form that you are experiencing is the result of the combined effort of the whole community, thanks to everyone who forced me to implement and discuss their suggestions, especially [@opusforlife2](https://github.com/opusforlife2) and [@B0pol](https://github.com/B0pol).
```

Reviewer: I have no issue with that. I just want the tense to remain consistent. You could change the whole para to first person, then.

## Conclusion

That’s it for this release. A big thanks goes out to all the people who made this possible, especially those not mentioned: Developers for helping out fellow developers, testers for testing, maintainers for maintaining, translators for translating and users like you for reporting the bugs in the first place. If you'd like to get a more technical view into the app, you can find the release page [here](https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.19.0) and the corresponding PR [here](https://github.com/TeamNewPipe/NewPipe/pull/3267).

> Today, I want to especially point out our [IRC channel](https://webchat.freenode.net/?channels=newpipe). Feel free to join our welcoming community in these trying times. Just to chat with all of us, discuss NewPipe, uptimes of servers, Ubuntu derivation or apparently floppy disks. It is free, easy and fun (I promise).

Reviewer: ```suggestion
Today, I want to especially point out our [IRC channel](https://webchat.freenode.net/?channels=newpipe). Feel free to join our welcoming community in these trying times, just to chat with all of us, or discuss NewPipe, uptimes of servers, Ubuntu derivation and even floppy disks, apparently. It is free, easy and fun (I promise).
```

> I also would love to shine some light on a big step in our translating area: Nepali is fully translated. Good job everybody involved in there. If you want to check out how you can help, head over to [Weblate](https://hosted.weblate.org/projects/newpipe/).

Reviewer: ```suggestion
I also would love to shine some light on a big step in our translating area: Nepali is fully translated. Good job everybody involved in that. If you want to check out how you can help, head over to [Weblate](https://hosted.weblate.org/projects/newpipe/).
```
